test/playback.test.js

```diff
@@ -165,3 +165,24 @@ QUnit.test('Live DASH', function(assert) {
     type: 'application/dash+xml'
   });
 });
+
+QUnit.test('loops', function(assert) {
+  assert.timeout(5000);
+
+  let done = assert.async();
+  let player = this.player;
+
+  player.loop(true);
+  player.src({
+    src: 'http://d2zihajmogu5jn.cloudfront.net/bipbop-advanced/bipbop_16x9_variant.m3u8',
+    type: 'application/x-mpegURL'
+  });
+  player.one('playing', function() {
+    player.vhs.mediaSource.on('sourceopen', () => {
```

Comment:
Is it possible for the playing event to fire before the media source is opened? If so, it may be best for us to add one more listener (for ended), then check for sourceopen after that.